pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotTaskManager.java

```diff
@@ -478,4 +489,34 @@ public void cleanUpTask() {
   public Scheduler getScheduler() {
     return _scheduledExecutorService;
   }
+
+  public synchronized void reportMetrics(String taskType) {
+    Map<String, TaskState> taskStates = _helixTaskResourceManager.getTaskStates(taskType);
+    Map<TaskState, Integer> taskStateToCountMap = new HashMap<>();
+    for (String taskName : taskStates.keySet()) {
+      TaskState taskState = taskStates.get(taskName);
+      if (!taskStateToCountMap.containsKey(taskState)) {
+        taskStateToCountMap.put(taskState, 0);
+      }
+      taskStateToCountMap.put(taskState, taskStateToCountMap.get(taskState) + 1);
+    }
+    // Reset all the status to 0
+    for (TaskState taskState : _taskStateToCountMap.keySet()) {
```

Comment:
No need to maintain the member `_taskStateToCountMap`, the local map should be enough.

Reply:
it's not, as we need to update the already reported metrics.
e.g.
in run 1, we report:
```
pinot_controller_taskStatus_Value{status="IN_PROGRESS",taskType="SegmentGenerationAndPushTask",} 0.0
```
in run 2, we need report:
```
pinot_controller_taskStatus_Value{status="IN_PROGRESS",taskType="SegmentGenerationAndPushTask",} 0.0
pinot_controller_taskStatus_Value{status="COMPLETED",taskType="SegmentGenerationAndPushTask",} 1.0
```
Without tracking the previous metrics, we cannot reset the value.

Reply:
I see, make sense